pkg/synthetic/ticket.go

```diff
@@ -0,0 +1,186 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package synthetic generates fake data for testing.
+package synthetic
+
+import (
+	"fmt"
+	"math"
+	"math/rand"
+
+	structpb "github.com/golang/protobuf/ptypes/struct"
+	"open-match.dev/open-match/pkg/pb"
+	"open-match.dev/open-match/pkg/structs"
+)
+
+// Tickets generates artificial tickets with these properties:
+// mmr.rating = value 0 to 100, average 50, standard deviation of 15.
+// one of the role properties:
+//   role.dps = 1 with 40% chance
+//   role.support = 1 with 33% chance
+//   role.tank = 1 with 27% chance
+// mode.demo = 1 to 5 (inclusive) with decreasing percentage each mode.
+// region.europe-east1 = latency number from 0ms to ~300ms
+// region.europe-west1 = ^
+// region.europe-west2 = ^  (all of these latency numbers are really
+// region.europe-west3 = ^   bad and don't represent real world data)
+// region.europe-west4 = ^
+func Tickets(count int) []*pb.Ticket {
+	result := make([]*pb.Ticket, count)
+	for i := 0; i < count; i++ {
+		result[i] = CustomTicket(defaultGenerators...)
+	}
+	return result
+}
+
```

Comment:
Whats the recommended usage for this library? Should this just be an external input? Do you expect callers to edit this file to add their own generators? If yes, how would a caller invoke a specific generator?

Reply:
For common generators, they can use the functions here.  If they have a unique generator, they can write their own.